aziot/src/internal/check/checks/host_local_time.rs

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+use super::prelude::*;
+
+#[derive(Serialize, Default)]
+pub struct HostLocalTime {
+    offset: Option<i64>,
+}
+
+#[async_trait::async_trait]
+impl Checker for HostLocalTime {
+    fn meta(&self) -> CheckerMeta {
+        CheckerMeta {
+            id: "host-local-time",
+            description: "host time is close to reference time",
+        }
+    }
+
+    async fn execute(&mut self, shared: &CheckerShared, cache: &mut CheckerCache) -> CheckResult {
+        self.execute_inner(shared, cache)
+            .await
+            .unwrap_or_else(CheckResult::Failed)
+    }
+}
+
+impl HostLocalTime {
+    async fn execute_inner(
+        &mut self,
+        shared: &CheckerShared,
+        _cache: &mut CheckerCache,
+    ) -> Result<CheckResult> {
+        // TODO: check against parent time in nested edge scenarios
```

Comment:
IS doesn't have any knowledge of nested edge so far. Edge does have awareness of the parent_hostname configuration and hence nested edge.

Reply:
Cool, so I can just remove this TODO then?

Reply:
Ahh, well, now I remember why I left this TODO... I accidentally liked to an older version of the check from before nested-edge. The latest version _does_ include some extra logic in nested-edge scenarios:
https://github.com/Azure/iotedge/blob/release/1.2/edgelet/iotedge/src/check/checks/host_local_time.rs#L41-L46

So, what should we do to account for that extra logic?

Reply:
The nested check seems strange because the root node of the nested edge topology would still need to be in sync with NTP (to connect to IoT Hub). All child nodes down the chain would then be transitively in sync with NTP. IS does not have the concept of `parent_hostname` because the parent hostname is the `iothub_hostname`. So, the NTP check is good enough for both cases in IS and you can remove the parent_hostname time check altogether.

Reply:
Awesome, that's what I like to hear :smile:

Reply:
We need both the NTP check and the HTTP Date header check in IS. If IS (and thus the check) doesn't know it's running in a nested scenario, the check might have to try NTP and then fall back to the HTTP Date header method.

@massand I don't understand what you mean by "all child nodes down the chain would then be transitively in sync with NTP."